### ISSUE TITLE
Readme Content Verification - need to format readmes to match template doc

### ISSUE THREAD

**Issue by scbedd** [Azure/azure-sdk-for-net#5499]
The following data-plane items need to be addressed. Validation has failed on the following readme list items. 

tl;dr: Readmes should have at least have sections that correspond to the main sections as defined in the [template readme.](https://github.com/scbedd/azure-sdk/blob/master/docs/README-TEMPLATE.md) 

This is what the **tool checks**, but the ideal scenario is the readme is reworked to be somewhat like the [readme example.](https://github.com/scbedd/azure-sdk/blob/master/docs/README-EXAMPLE.md)

For a rundown on what you need to do to resolve these breaking issues ASAP, check out aka.ms/azure-sdk-analyze-failed

## Content Validation Items
23 readmes have at least one missing required section. Run `ward scan -d <repo root> -c <repo root>/eng/.docsettings.yml` to get the full list.

- [ ] README.md
- [ ] doc/README.md
- [ ] sdk/core/Azure.Core/README.md
- [ ] sdk/core/Microsoft.Extensions.Azure/README.md
- [ ] sdk/keyvault/Microsoft.Azure.KeyVault/README.md
- [ ] sdk/servicebus/Microsoft.Azure.ServiceBus/README.md
- [ ] sdk/eventhub/Microsoft.Azure.EventHubs/README.md
- [ ] sdk/alertsmanagement/Microsoft.Azure.Management.AlertsManagement/README.md
- [ ] sdk/appconfiguration/README.md
- [ ] sdk/appconfiguration/Azure.Data.AppConfiguration/tests/Readme.md
- [ ] sdk/appconfiguration/Azure.Data.AppConfiguration/samples/README.md
- [ ] sdk/core/Azure.Core/README.md
- [ ] sdk/storage/README.md
- [ ] sdk/storage/Azure.Storage.Blobs/swagger/readme.md
- [ ] sdk/storage/Azure.Storage.Files.Shares/swagger/readme.md
- [ ] sdk/storage/Azure.Storage.Files.DataLake/swagger/readme.md
- [ ] sdk/storage/Azure.Storage.Queues/swagger/readme.md
- [ ] sdk/storage/Azure.Storage.Common/swagger/Generator/readme.md
- [ ] sdk/cognitiveservices/Language.TextAnalytics/src/Readme.md
- [ ] sdk/cognitiveservices/Personalizer/src/Readme.md
- [ ] sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/README.md
- [ ] sdk/batch/Microsoft.Azure.Batch/README.md
- [ ] sdk/batch/Microsoft.Azure.Batch.Conventions.Files/README.md

## Presence Validation  Items
Warden detected there to be a missing readme in the below relative locations. These are much squishier than content verification, as there are real reasons for readmes not being present. Getting eyes on what `doc-warden` is ignoring is still good for awareness though. If a readme "missing" below is correct, feel free to remove the exception from the checklist.

For the .NET repo, it is very difficult to tell what is actually missing, and what is missing due to repo structure. However, @mmacy was curious what we saw missing for .NET. The detected missing ones are below.

32 missing readmes detected at:
- [ ] sdk/eventgrid/Microsoft.Azure.EventGrid/README.md
- [ ] sdk/operationalinsights/Microsoft.Azure.OperationalInsights/README.md
- [ ] sdk/hdinsight/Microsoft.Azure.HDInsight.Job/README.md
- [ ] sdk/alertsmanagement/Microsoft.Azure.Management.AlertsManagement/README.md
- [ ] sdk/applicationinsights/Microsoft.Azure.ApplicationInsights.Query/README.md
- [ ] sdk/containerregistry/Microsoft.Azure.ContainerRegistry/README.md
- [ ] sdk/cognitiveservices/Language.LUIS.Runtime/README.md
- [ ] sdk/cognitiveservices/Search.BingEntitySearch/README.md
- [ ] sdk/cognitiveservices/Search.BingWebSearch/README.md
- [ ] sdk/cognitiveservices/Knowledge.QnAMaker/README.md
- [ ] sdk/cognitiveservices/AnomalyDetector/README.md
- [ ] sdk/cognitiveservices/Search.BingVisualSearch/README.md
- [ ] sdk/cognitiveservices/Search.BingCustomSearch/README.md
- [ ] sdk/cognitiveservices/Search.BingImageSearch/README.md
- [ ] sdk/cognitiveservices/Vision.FormRecognizer/README.md
- [ ] sdk/cognitiveservices/Search.BingVideoSearch/README.md
- [ ] sdk/cognitiveservices/Search.BingNewsSearch/README.md
- [ ] sdk/cognitiveservices/Personalizer/README.md
- [ ] sdk/cognitiveservices/Vision.ComputerVision/README.md
- [ ] sdk/cognitiveservices/Vision.CustomVision.Training/README.md
- [ ] sdk/cognitiveservices/Vision.Face/README.md
- [ ] sdk/cognitiveservices/Search.BingAutoSuggest/README.md
- [ ] sdk/cognitiveservices/Language.LUIS.Authoring/README.md
- [ ] sdk/cognitiveservices/Vision.ContentModerator/README.md
- [ ] sdk/cognitiveservices/Vision.CustomVision.Prediction/README.md
- [ ] sdk/cognitiveservices/Language.SpellCheck/README.md
- [ ] sdk/cognitiveservices/Language.TextAnalytics/README.md
- [ ] sdk/cognitiveservices/Search.BingLocalSearch/README.md
- [ ] sdk/cognitiveservices/Search.BingCustomImageSearch/README.md
- [ ] sdk/cognitiveservices/FormRecognizer/README.md
- [ ] sdk/batch/Microsoft.Azure.Batch.FileStaging/README.md
- [ ] sdk/graphrbac/Microsoft.Azure.Graph.RBAC/README.md

**Comment by jongio** [comment]
@scbedd @AlexGhiondea - Does this need to stay open or close?

**Comment by annelo-msft** [comment]
Issues related to opt-out include:
- Verify only for package READMEs: https://github.com/Azure/azure-sdk-tools/issues/42
- Request to opt-out of sections: https://github.com/Azure/azure-sdk-tools/issues/404
- Remove README exclusions: https://github.com/Azure/azure-sdk-for-net/issues/15423

**Comment by annelo-msft** [comment]
@scbedd, I'm not getting any output for the `ward` command listed in the description.  If there's any remaining work associated with this issue, please let us know.  Thanks!

**Comment by scbedd** [comment]
@annelo-msft I'm slow on notifications 👎 

Are you invoking it as the CI does? If so, you're likely passing along the [`.docsettings.yml`](https://github.com/Azure/azure-sdk-for-net/blob/main/eng/.docsettings.yml#L37) that explicitly silences a lot of these issues.

Reasoning was that we didn't want to turn all pipelines red forever. So we filed the issues for backlog address and moved on from there.

To see the issues as they are now, just remove the `known-X` exceptions from the .docsettings before running.

All that being said, we've lived long enough without traction on this issue, is it worth re-opening? 

Up to you, but that's the full context! 👍
